cql-parser/src/main/scala/troy/cql/parser/CqlParser.scala

```diff
@@ -53,7 +55,20 @@ object CqlParser extends JavaTokenParsers
    */
   def constant: Parser[Constant] = {
     import Constants._
-    (string | uuid | number | boolean) ^^ Constant // | hex // TODO
```

Comment:
Let's restore this line, and move the mapping logic inside to each matcher separately. 
The only problem you will face is that Constants.string is used in different places as Parser[String] not Parser[StringConstant], so you may define another version outside `Constants` for other usages.